unittests/AST/ASTImporterTest.cpp

```diff
@@ -2543,5 +2543,60 @@ long b = strtol(&a);
   EXPECT_EQ(DeclCounter<UsingShadowDecl>().match(ToTU, usingShadowDecl()), 1u);
 }
 
+TEST_F(Fixture, MakeInjectedClassNameTypeWithPrevDecl) {
```

Comment:
I wonder if unit test is the best way to to trigger this issue. What about an ASTMerge regression test?

Reply:
I could not easily produce a more simple way to get the original assertion.

Reply:
What is the difference between ASTMerge and Import tests (where to put a new test)?

Reply:
The test cases `check-clang-astmerge` are using the Importer directly in a intuitive way (See `lib/frontend/ASTMerge.cpp`). I'd suggest to write a test with these if you really want lit tests. For me the unit tests we have is perfectly fine.

 The other lit tests are in `check-clang-import` which uses the `clang-import-test` binary. And this uses the Importer indirectly through `ExternalASTMerger`. If I understand correctly then this tool does allow to import incomplete Decls and DeclContexts, while the other (`check-clang-astmerge`) does not. 
So we can merge these two files:
```
enum E {
  a = 1,
  b = 2
};
```
```
void expr() {
  static_assert(E::a + E::b == 3);
}
```
However with the first tool `clang -ast-merge` that would not be possible because the second TU would not compile (since E has a missing definition).